example/babelstream/src/AlpakaStream.cpp

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2015-16 Tom Deakin, Simon McIntosh-Smith,
+// University of Bristol HPC
+//
+// For full license terms please see the LICENSE file distributed with this
+// source code
+//
+// Cupla version created by Jeff Young in 2021
+// Ported from cupla to alpaka by Bernhard Manfred Gruber in 2022
+
+#include "AlpakaStream.h"
+
+constexpr auto TBSIZE = 1024;
```

Comment:
In vikunja, the block size is also [hard coded](https://github.com/alpaka-group/vikunja/blob/d6cbf89d783cde91fe785eec4192142244c4e0e8/include/vikunja/reduce/reduce.hpp#L120) for the moment. For example [CUDA use 256](https://github.com/alpaka-group/vikunja/blob/d6cbf89d783cde91fe785eec4192142244c4e0e8/include/vikunja/workdiv/BlockBasedWorkDiv.hpp#L84). I suggest to test one run with a smaller block size. If you get a speed up, think about a possibility to set the block size via compile time parameter.

Reply:
There is not a big difference between the block sizes:
```
BabelStream
Version: 4.0
Implementation: alpaka
Running kernels 100 times
Precision: double
Array size: 268.4 MB (=0.3 GB)
Total size: 805.3 MB (=0.8 GB)
Using alpaka device Quadro GV100

with 1024:
Function    MBytes/sec  Min (sec)   Max         Average
Copy        590977.242  0.00091     0.00096     0.00091
Mul         587894.898  0.00091     0.00094     0.00092
Add         615253.841  0.00131     0.00133     0.00131
Triad       614809.491  0.00131     0.00133     0.00132
Dot         644256.949  0.00083     0.00087     0.00084

with 512:
Function    MBytes/sec  Min (sec)   Max         Average
Copy        592219.119  0.00091     0.00091     0.00091
Mul         588827.263  0.00091     0.00092     0.00092
Add         616218.491  0.00131     0.00133     0.00131
Triad       615375.609  0.00131     0.00132     0.00131
Dot         642840.616  0.00084     0.00086     0.00084

with 256:
Function    MBytes/sec  Min (sec)   Max         Average
Copy        592287.720  0.00091     0.00091     0.00091
Mul         589497.085  0.00091     0.00092     0.00092
Add         616160.499  0.00131     0.00132     0.00131
Triad       615247.261  0.00131     0.00132     0.00131
Dot         649585.728  0.00083     0.00085     0.00083
```
But we need to revisit this anyway to support CPU devices as well. But this will be a follow-up PR.

Reply:
Okay, that's fine.